### PR TITLE
fix: update the application even if gitop fails

### PR DESCRIPTION
**Pull request by stuartwdouglas**
When deleting a component currently the Application will not be updated if the gitops deletion fails.

fixes [RHTAPBUGS-503](https://issues.redhat.com//browse/RHTAPBUGS-503)

### What does this PR do?:

Currently the finalizer for the component cleans up gitops and removes the component from the application devfile, however if the gitops fails the application is not updated.

Note that I am not 100% sure that this is the correct behavior, but it seems to make sense.

### Which issue(s)/story(ies) does this PR fixes:

https://issues.redhat.com/browse/RHTAPBUGS-503

### CODE REVIEW

**Review by johnmcollier — APPROVED** [review]

**Comment by stuartwdouglas** [comment]
Is anything holding this up?

**Comment by johnmcollier** [comment]
@stuartwdouglas Hey sorry, forgot that you didn't have merge rights on this repo, let's get this in

**Pull request merged**
